blockchain/vm/contracts.go

```diff
@@ -116,33 +116,55 @@ var PrecompiledContractsKore = map[common.Address]PrecompiledContract{
 	common.BytesToAddress([]byte{3, 255}): &validateSender{},
 }
 
+// PrecompiledContractsCancun contains the default set of pre-compiled Klaytn
+// Have the same list with Kore
+var PrecompiledContractsCancun = map[common.Address]PrecompiledContract{
+	common.BytesToAddress([]byte{1}):      &ecrecover{},
+	common.BytesToAddress([]byte{2}):      &sha256hash{},
+	common.BytesToAddress([]byte{3}):      &ripemd160hash{},
+	common.BytesToAddress([]byte{4}):      &dataCopy{},
+	common.BytesToAddress([]byte{5}):      &bigModExp{eip2565: true},
+	common.BytesToAddress([]byte{6}):      &bn256AddIstanbul{},
+	common.BytesToAddress([]byte{7}):      &bn256ScalarMulIstanbul{},
+	common.BytesToAddress([]byte{8}):      &bn256PairingIstanbul{},
+	common.BytesToAddress([]byte{9}):      &blake2F{},
+	common.BytesToAddress([]byte{3, 253}): &vmLog{},
+	common.BytesToAddress([]byte{3, 254}): &feePayer{},
+	common.BytesToAddress([]byte{3, 255}): &validateSender{},
+}
+
 var (
-	PrecompiledAddressesIstanbulCompatible  []common.Address
-	PrecompiledAddressesByzantiumCompatible []common.Address
+	PrecompiledAddressCancun      []common.Address
+	PrecompiledAddressIstanbul    []common.Address
+	PrecompiledAddressesByzantium []common.Address
 )
 
 func init() {
 	for k := range PrecompiledContractsByzantiumCompatible {
-		PrecompiledAddressesByzantiumCompatible = append(PrecompiledAddressesByzantiumCompatible, k)
+		PrecompiledAddressesByzantium = append(PrecompiledAddressesByzantium, k)
 	}
 
 	// After istanbulCompatible hf, need to support for vmversion0 contracts, too.
 	// VmVersion0 contracts are deployed before istanbulCompatible and they use byzantiumCompatible precompiled contracts.
 	// VmVersion0 contracts are the contracts deployed before istanbulCompatible hf.
 	for k := range PrecompiledContractsIstanbulCompatible {
-		PrecompiledAddressesIstanbulCompatible = append(PrecompiledAddressesIstanbulCompatible, k)
+		PrecompiledAddressIstanbul = append(PrecompiledAddressIstanbul, k)
+	}
+
+	for k := range PrecompiledContractsIstanbulCompatible {
+		PrecompiledAddressCancun = append(PrecompiledAddressCancun, k)
 	}
-	PrecompiledAddressesIstanbulCompatible = append(PrecompiledAddressesIstanbulCompatible,
-		[]common.Address{common.BytesToAddress([]byte{10}), common.BytesToAddress([]byte{11})}...)
 }
 
 // ActivePrecompiles returns the precompiles enabled with the current configuration.
 func ActivePrecompiles(rules params.Rules) []common.Address {
 	switch {
+	case rules.IsCancun:
+		return PrecompiledAddressCancun
 	case rules.IsIstanbul:
-		return PrecompiledAddressesIstanbulCompatible
+		return PrecompiledAddressIstanbul
 	default:
-		return PrecompiledAddressesByzantiumCompatible
+		return PrecompiledAddressesByzantium
```

Comment:
`10`, `11` addresses needs to be added to AccessList since those addresses can be accessed if the contract is deployed before istanbul hardfork (vmVersion==0). What about next lines? Could you add those addresses in `ActivePrecompileds`?
```
// ActivePrecompiles returns the precompiles enabled with the current configuration.
func ActivePrecompiles(rules params.Rules) []common.Address {
	var precompiledAddresses []common.Address
	switch {
	case rules.IsCancun:
		precompiledAddresses = append(precompiledAddresses, PrecompiledAddressesCancunCompatible...)
	case rules.IsIstanbul:
		precompiledAddresses = append(precompiledAddresses, PrecompiledAddressesIstanbulCompatible...)
	default:
		precompiledAddresses = append(precompiledAddresses, PrecompiledAddressesByzantiumCompatible...)
	}
	// After istanbulCompatible hf, need to support for vmversion0 contracts, too.
	// VmVersion0 contracts are deployed before istanbulCompatible and they use byzantiumCompatible precompiled contracts.
	// VmVersion0 contracts are the contracts deployed before istanbulCompatible hf.
	if rules.IsIstanbul {
		precompiledAddresses = append(precompiledAddresses,
			[]common.Address{common.BytesToAddress([]byte{10}), common.BytesToAddress([]byte{11})}...)
	}
	return precompiledAddresses
}
```

Reply:
Following the Kore HF, addresses {10,12} were excluded from loading due to Kore's precompile map. Initially, I considered removing this exclusion since the HF occurred a year ago. However, I now prioritize consistency over time elapsed. The main difference in your suggestion is the condition related to the HF.

So, how about this fix?
```
if rules.IsIstanbul {
		precompiledAddresses = append(precompiledAddresses,
			[]common.Address{common.BytesToAddress([]byte{10}), common.BytesToAddress([]byte{11})}...)
	}
```

Reply:
> Following the Kore HF, addresses {10,12} were excluded from loading due to Kore's precompile map.

Is this about `ActivePrecompile`? As far as I know, `IsIstanbul` is true even the block number passes kore hardfork. So, I think those addresses are still included in kore precompiled addresses. If I undertood wrong, please let me know.

Reply:
Yes right. I was confused that the two variables were the same.. Sorry for the noise. Also, the rule condition `IsIstanbul` seems enough rather than `IsCancun`. PTAl. Thank you